### ISSUE TITLE
Write library for pushing to Gogs

### ISSUE THREAD

**Issue by jag3773** [unfoldingWord-dev/ts-android#1259]
We need to start pushing content into gogs, which is at git.door43.org. The docs for the gogs API are at https://github.com/gogits/go-gogs-client/wiki.

This will require at least the following:

- [x] A way to create an account on git.door43.org (if needed)
- [x] A way to sign into said account
- [x] A way to associate the SSH key of the device with the user's account
- [x] A way to associate the local repo with the remote repo, which would have the form of [useraccount]/[reponame]
- [x] Pushing probably already built and may not need to change

Same as https://github.com/unfoldingWord-dev/ts-desktop/issues/304#issue-131387504

**Comment by neutrinog** [comment]
Blocked by #887 for security reasons

**Comment by RRohrssen** [comment]
The interface for resolving merge conflicts is already done: #1255

**Comment by neutrinog** [comment]
@RRohrssen You didn't read far enough in that commit ;) UI for merge conflicts hasn't been completed yet. Only the merging part has been completed. see #1290

**Comment by RRohrssen** [comment]
@neutrinog Thanks. I modified the comment.

**Comment by RRohrssen** [comment]
@neutrinog, @PhotoNomad0 which (if any) of the main parts of this issue are completed? I converted the initial issue to checkboxes. Please check those off and add issue numbers to any closed or open issues that correspond.

My suspicion is that each of these is a NEW issue. I didn't find issues for the corresponding steps.

**Comment by neutrinog** [comment]
@RRohrssen none of these are complete yet. I'm actually starting on this today.

**Comment by neutrinog** [comment]
@richmahn I need help! The API hates me!

**Comment by neutrinog** [comment]
blocked by #1250

**Comment by neutrinog** [comment]
a library for accessing the api has been written for this. The library is at https://github.com/unfoldingWord-dev/android-gogs-client and has already been configured in this project. We just need to use it when ready.
